src/interfaces.ts

```diff
@@ -172,6 +172,79 @@ export enum Event {
   RemoteBookmark = 'remote-bookmark',
 }
 
+export interface EventListenerProp {
+  [Event.PlaybackState]: {
+    state: State
+  }
+  [Event.PlaybackError]: {
+    code: string
+    message: string
+  }
+  [Event.PlaybackQueueEnded]: {
+    track: number
+    position: number
+  }
+  [Event.PlaybackTrackChanged]: {
+    track: number
+    position: number
+    nextTrack: number
+  }
+  [Event.PlaybackMetadataReceived]: {
+    source: string
+    title: string | null
+    url: string | null
+    artist: string | null
+    album: string | null
+    date: string | null
+    genre: string | null
+  }
+  [Event.PlaybackProgressUpdated]: {
```

Comment:
Please see the dedicated interface that already exists for this (`ProgressUpdateEvent`). More generally, please abstract all the event structures into their own interfaces and use them in this typemap.